2.4.3修改为Postgresql数据库后，更新配置报错

版本：2.4.3
使用Postgresql数据库、启动正常，更新配置时报错：

commands ignored until end of transaction block;nested exception is org.postgresql.util.PSQLException: ERROR: current transaction is aborted,
commands ignored until end of transaction block;caused: ERROR: current transaction is aborted,
commands ignored until end of transaction block;caused: ERROR: function now(integer)does not exist Hint: No function matches the given name and argument types.You might need to add explicit type casts.Position: 81;

遇到同样的问题，社区有相关解决方案吗？

./config-fatal.log:2024-12-13 08:28:33,961 ERROR [db-error] try to update or add config failed, PreparedStatementCallback; bad SQL grammar []; nested exception is org.postgresql.util.PSQLException: ERROR: function now(integer) does not exist
./config-fatal.log:2024-12-13 08:35:47,829 ERROR [db-error] try to update or add config failed, PreparedStatementCallback; bad SQL grammar []; nested exception is org.postgresql.util.PSQLException: ERROR: function now(integer) does not exist
./config-server.log:2024-12-13 00:04:02,832 ERROR An error occurred while executing method [removeConfigHistory].
./config-server.log:org.springframework.jdbc.BadSqlGrammarException: PreparedStatementCallback; bad SQL grammar [DELETE FROM his_config_info WHERE gmt_modified < ? LIMIT ?]; nested exception is org.postgresql.util.PSQLException: ERROR: syntax error at or near "LIMIT"
./config-server.log:Caused by: org.postgresql.util.PSQLException: ERROR: syntax error at or near "LIMIT"

--- a/nacos-datasource-plugin-ext/nacos-datasource-plugin-ext-base/src/main/java/com/alibaba/nacos/plugin/datasource/dialect/DatabaseDialect.java
+++ b/nacos-datasource-plugin-ext/nacos-datasource-plugin-ext-base/src/main/java/com/alibaba/nacos/plugin/datasource/dialect/DatabaseDialect.java
@@ -16,6 +16,8 @@
 
 package com.alibaba.nacos.plugin.datasource.dialect;
 
+import com.alibaba.nacos.plugin.datasource.mapper.Mapper;
+
 /**
  * DatabaseDialect interface.
  * @author Long Yu
@@ -82,4 +84,12 @@ public interface DatabaseDialect {
      */
     public String[] getReturnPrimaryKeys();
 
+    /**
+     * Get the function corresponding to the dialect according to the function name
+     * @author Mr.Muzhi
+     * @since 2025/1/7 16:30
+     * @param functionName functionName
+     * @return function
+     */
+    String getFunction(String functionName);
 }

--- a/nacos-datasource-plugin-ext/nacos-datasource-plugin-ext-base/src/main/java/com/alibaba/nacos/plugin/datasource/dialect/DefaultDatabaseDialect.java
+++ b/nacos-datasource-plugin-ext/nacos-datasource-plugin-ext-base/src/main/java/com/alibaba/nacos/plugin/datasource/dialect/DefaultDatabaseDialect.java
@@ -17,6 +17,7 @@
 package com.alibaba.nacos.plugin.datasource.dialect;
 
 import com.alibaba.nacos.plugin.datasource.constants.DatabaseTypeConstant;
+import com.alibaba.nacos.plugin.datasource.enums.mysql.TrustedMysqlFunctionEnum;
 
 /**
  * defauLT database dialect.
@@ -27,6 +28,11 @@ public class DefaultDatabaseDialect extends AbstractDatabaseDialect {
     @Override
     public String getType() {
         return DatabaseTypeConstant.MYSQL;
+    }
+
+    @Override
+    public String getFunction(String functionName) {
+        return TrustedMysqlFunctionEnum.getFunctionByName(functionName);
     }
 
 }

--- a/nacos-datasource-plugin-ext/nacos-datasource-plugin-ext-base/src/main/java/com/alibaba/nacos/plugin/datasource/impl/base/BaseConfigInfoAggrMapper.java
+++ b/nacos-datasource-plugin-ext/nacos-datasource-plugin-ext-base/src/main/java/com/alibaba/nacos/plugin/datasource/impl/base/BaseConfigInfoAggrMapper.java
@@ -59,5 +59,9 @@ public class BaseConfigInfoAggrMapper extends ConfigInfoAggrMapperByMySql {
         List<Object> paramList = CollectionUtils.list(dataId, groupId, tenantId);
         return new MapperResult(sql, paramList);
     }
-    
+
+    @Override
+    public String getFunction(String functionName) {
+        return databaseDialect.getFunction(functionName);
+    }
 }

--- a/nacos-datasource-plugin-ext/nacos-datasource-plugin-ext-base/src/main/java/com/alibaba/nacos/plugin/datasource/impl/base/BaseConfigInfoBetaMapper.java
+++ b/nacos-datasource-plugin-ext/nacos-datasource-plugin-ext-base/src/main/java/com/alibaba/nacos/plugin/datasource/impl/base/BaseConfigInfoBetaMapper.java
@@ -59,6 +59,11 @@ public class BaseConfigInfoBetaMapper extends ConfigInfoBetaMapperByMySql {
                         + " FROM ( " + sqlInner + "  )" + "  g, config_info_beta t WHERE g.id = t.id ";
         return new MapperResult(sql, Collections.emptyList());
     }
+
+    @Override
+    public String getFunction(String functionName) {
+        return databaseDialect.getFunction(functionName);
+    }
     
     
 }

--- a/nacos-datasource-plugin-ext/nacos-datasource-plugin-ext-base/src/main/java/com/alibaba/nacos/plugin/datasource/impl/base/BaseConfigInfoMapper.java
+++ b/nacos-datasource-plugin-ext/nacos-datasource-plugin-ext-base/src/main/java/com/alibaba/nacos/plugin/datasource/impl/base/BaseConfigInfoMapper.java
@@ -284,5 +284,9 @@ public class BaseConfigInfoMapper extends ConfigInfoMapperByMySql {
     public String getTableName() {
         return TableConstant.CONFIG_INFO;
     }
-    
+
+    @Override
+    public String getFunction(String functionName) {
+        return databaseDialect.getFunction(functionName);
+    }
 }

--- a/nacos-datasource-plugin-ext/nacos-datasource-plugin-ext-base/src/main/java/com/alibaba/nacos/plugin/datasource/impl/base/BaseConfigInfoTagMapper.java
+++ b/nacos-datasource-plugin-ext/nacos-datasource-plugin-ext-base/src/main/java/com/alibaba/nacos/plugin/datasource/impl/base/BaseConfigInfoTagMapper.java
@@ -53,5 +53,10 @@ public class BaseConfigInfoTagMapper extends ConfigInfoTagMapperByMySql {
                 + innerSql + "  ) " + "g, config_info_tag t  WHERE g.id = t.id  ";
         return new MapperResult(sql, Collections.emptyList());
     }
+
+    @Override
+    public String getFunction(String functionName) {
+        return databaseDialect.getFunction(functionName);
+    }
     
 }

--- a/nacos-datasource-plugin-ext/nacos-datasource-plugin-ext-base/src/main/java/com/alibaba/nacos/plugin/datasource/impl/base/BaseConfigTagsRelationMapper.java
+++ b/nacos-datasource-plugin-ext/nacos-datasource-plugin-ext-base/src/main/java/com/alibaba/nacos/plugin/datasource/impl/base/BaseConfigTagsRelationMapper.java
@@ -144,5 +144,10 @@ public class BaseConfigTagsRelationMapper extends ConfigTagsRelationMapperByMySq
         String sql = getLimitPageSqlWithOffset(sqlFetchRows + where, startRow, pageSize);
         return new MapperResult(sql, paramList);
     }
+
+    @Override
+    public String getFunction(String functionName) {
+        return databaseDialect.getFunction(functionName);
+    }
     
 }

--- a/nacos-datasource-plugin-ext/nacos-datasource-plugin-ext-base/src/main/java/com/alibaba/nacos/plugin/datasource/impl/base/BaseGroupCapacityMapper.java
+++ b/nacos-datasource-plugin-ext/nacos-datasource-plugin-ext-base/src/main/java/com/alibaba/nacos/plugin/datasource/impl/base/BaseGroupCapacityMapper.java
@@ -43,5 +43,10 @@ public class BaseGroupCapacityMapper extends GroupCapacityMapperByMysql {
         return new MapperResult(sql,
                 CollectionUtils.list(context.getWhereParameter(FieldConstant.ID), context.getPageSize()));
     }
+
+    @Override
+    public String getFunction(String functionName) {
+        return databaseDialect.getFunction(functionName);
+    }
     
 }

--- a/nacos-datasource-plugin-ext/nacos-datasource-plugin-ext-base/src/main/java/com/alibaba/nacos/plugin/datasource/impl/base/BaseTenantCapacityMapper.java
+++ b/nacos-datasource-plugin-ext/nacos-datasource-plugin-ext-base/src/main/java/com/alibaba/nacos/plugin/datasource/impl/base/BaseTenantCapacityMapper.java
@@ -43,5 +43,10 @@ public class BaseTenantCapacityMapper extends TenantCapacityMapperByMySql {
         return new MapperResult(sql, CollectionUtils.list(context.getWhereParameter(FieldConstant.ID),
                 context.getWhereParameter(FieldConstant.LIMIT_SIZE)));
     }
+    @Override
+    public String getFunction(String functionName) {
+        return databaseDialect.getFunction(functionName);
+    }
+
 
 }

--- a/nacos-datasource-plugin-ext/nacos-datasource-plugin-ext-base/src/main/java/com/alibaba/nacos/plugin/datasource/impl/base/BaseTenantInfoMapper.java
+++ b/nacos-datasource-plugin-ext/nacos-datasource-plugin-ext-base/src/main/java/com/alibaba/nacos/plugin/datasource/impl/base/BaseTenantInfoMapper.java
@@ -16,7 +16,9 @@
 
 package com.alibaba.nacos.plugin.datasource.impl.base;
 
+import com.alibaba.nacos.plugin.datasource.dialect.DatabaseDialect;
 import com.alibaba.nacos.plugin.datasource.impl.mysql.TenantInfoMapperByMySql;
+import com.alibaba.nacos.plugin.datasource.manager.DatabaseDialectManager;
 
 /**
  * The base implementation of TenantInfo.
@@ -24,5 +26,13 @@ import com.alibaba.nacos.plugin.datasource.impl.mysql.TenantInfoMapperByMySql;
  * @author Long Yu
  **/
 public class BaseTenantInfoMapper extends TenantInfoMapperByMySql {
+
+    private DatabaseDialect databaseDialect;
+
+    public BaseTenantInfoMapper() {
+        databaseDialect = DatabaseDialectManager.getInstance().getDialect(getDataSource());
+    }
+
+
 
 }

--- a/nacos-datasource-plugin-ext/nacos-dm-datasource-plugin-ext/src/main/java/com/alibaba/nacos/plugin/datasource/dialect/DaMengDatabaseDialect.java
+++ b/nacos-datasource-plugin-ext/nacos-dm-datasource-plugin-ext/src/main/java/com/alibaba/nacos/plugin/datasource/dialect/DaMengDatabaseDialect.java
@@ -18,6 +18,7 @@ package com.alibaba.nacos.plugin.datasource.dialect;
 
 import com.alibaba.nacos.plugin.datasource.constants.DatabaseTypeConstant;
 import com.alibaba.nacos.plugin.datasource.constants.PrimaryKeyConstant;
+import com.alibaba.nacos.plugin.datasource.emums.TrustedDaMengFunctionEnum;
 
 /**
  * dameng database dialect.
@@ -34,6 +35,12 @@ public class DaMengDatabaseDialect extends AbstractDatabaseDialect {
     @Override
     public String getType() {
         return DatabaseTypeConstant.DM;
+    }
+
+
+    @Override
+    public String getFunction(String functionName) {
+        return TrustedDaMengFunctionEnum.getFunctionByName(functionName);
     }
 
 }

--- a/nacos-datasource-plugin-ext/nacos-dm-datasource-plugin-ext/src/main/java/com/alibaba/nacos/plugin/datasource/emums/TrustedDaMengFunctionEnum.java
+++ b/nacos-datasource-plugin-ext/nacos-dm-datasource-plugin-ext/src/main/java/com/alibaba/nacos/plugin/datasource/emums/TrustedDaMengFunctionEnum.java
@@ -1,0 +1,66 @@
+/*
+ * Copyright 1999-2018 Alibaba Group Holding Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.alibaba.nacos.plugin.datasource.emums;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * The TrustedSqlFunctionEnum enum class is used to enumerate and manage a list of trusted built-in SQL functions.
+ * By using this enum, you can verify whether a given SQL function is part of the trusted functions list
+ * to avoid potential SQL injection risks.
+ *
+ * @author blake.qiu
+ */
+public enum TrustedDaMengFunctionEnum {
+
+    /**
+     * NOW().
+     */
+    NOW("NOW()", "SYSDATE");
+
+    private static final Map<String, TrustedDaMengFunctionEnum> LOOKUP_MAP = new HashMap<>();
+
+    static {
+        for (TrustedDaMengFunctionEnum entry : TrustedDaMengFunctionEnum.values()) {
+            LOOKUP_MAP.put(entry.functionName, entry);
+        }
+    }
+
+    private final String functionName;
+
+    private final String function;
+
+    TrustedDaMengFunctionEnum(String functionName, String function) {
+        this.functionName = functionName;
+        this.function = function;
+    }
+
+    /**
+     * Get the function name.
+     *
+     * @param functionName function name
+     * @return function
+     */
+    public static String getFunctionByName(String functionName) {
+        TrustedDaMengFunctionEnum entry = LOOKUP_MAP.get(functionName);
+        if (entry != null) {
+            return entry.function;
+        }
+        throw new IllegalArgumentException(String.format("Invalid function name: %s", functionName));
+    }
+}

--- a/nacos-datasource-plugin-ext/nacos-mssql-datasource-plugin-ext/src/main/java/com/alibaba/nacos/plugin/datasource/dialect/SqlServerDatabaseDialect.java
+++ b/nacos-datasource-plugin-ext/nacos-mssql-datasource-plugin-ext/src/main/java/com/alibaba/nacos/plugin/datasource/dialect/SqlServerDatabaseDialect.java
@@ -17,6 +17,7 @@
 package com.alibaba.nacos.plugin.datasource.dialect;
 
 import com.alibaba.nacos.plugin.datasource.constants.DatabaseTypeConstant;
+import com.alibaba.nacos.plugin.datasource.enums.TrustedSqlServerFunctionEnum;
 
 /**
  * Microsoft SQL Server database dialect.
@@ -50,5 +51,10 @@ public class SqlServerDatabaseDialect extends AbstractDatabaseDialect {
     public String getLimitPageSqlWithOffset(String sql, int startOffset, int pageSize){
         return sql + " ORDER BY id OFFSET " + startOffset + " ROWS FETCH NEXT "
                 + pageSize + " ROWS ONLY ";
+    }
+
+    @Override
+    public String getFunction(String functionName) {
+        return TrustedSqlServerFunctionEnum.getFunctionByName(functionName);
     }
 }

--- a/nacos-datasource-plugin-ext/nacos-mssql-datasource-plugin-ext/src/main/java/com/alibaba/nacos/plugin/datasource/enums/TrustedSqlServerFunctionEnum.java
+++ b/nacos-datasource-plugin-ext/nacos-mssql-datasource-plugin-ext/src/main/java/com/alibaba/nacos/plugin/datasource/enums/TrustedSqlServerFunctionEnum.java
@@ -1,0 +1,66 @@
+/*
+ * Copyright 1999-2018 Alibaba Group Holding Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.alibaba.nacos.plugin.datasource.enums;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * The TrustedSqlFunctionEnum enum class is used to enumerate and manage a list of trusted built-in SQL functions.
+ * By using this enum, you can verify whether a given SQL function is part of the trusted functions list
+ * to avoid potential SQL injection risks.
+ *
+ * @author blake.qiu
+ */
+public enum TrustedSqlServerFunctionEnum {
+
+    /**
+     * NOW().
+     */
+    NOW("NOW()", "SYSDATETIME()");
+
+    private static final Map<String, TrustedSqlServerFunctionEnum> LOOKUP_MAP = new HashMap<>();
+
+    static {
+        for (TrustedSqlServerFunctionEnum entry : TrustedSqlServerFunctionEnum.values()) {
+            LOOKUP_MAP.put(entry.functionName, entry);
+        }
+    }
+
+    private final String functionName;
+
+    private final String function;
+
+    TrustedSqlServerFunctionEnum(String functionName, String function) {
+        this.functionName = functionName;
+        this.function = function;
+    }
+
+    /**
+     * Get the function name.
+     *
+     * @param functionName function name
+     * @return function
+     */
+    public static String getFunctionByName(String functionName) {
+        TrustedSqlServerFunctionEnum entry = LOOKUP_MAP.get(functionName);
+        if (entry != null) {
+            return entry.function;
+        }
+        throw new IllegalArgumentException(String.format("Invalid function name: %s", functionName));
+    }
+}

--- a/nacos-datasource-plugin-ext/nacos-oracle-datasource-plugin-ext/src/main/java/com/alibaba/nacos/plugin/datasource/dialect/OracleDatabaseDialect.java
+++ b/nacos-datasource-plugin-ext/nacos-oracle-datasource-plugin-ext/src/main/java/com/alibaba/nacos/plugin/datasource/dialect/OracleDatabaseDialect.java
@@ -18,6 +18,7 @@ package com.alibaba.nacos.plugin.datasource.dialect;
 
 import com.alibaba.nacos.common.utils.NamespaceUtil;
 import com.alibaba.nacos.plugin.datasource.constants.DatabaseTypeConstant;
+import com.alibaba.nacos.plugin.datasource.enums.TrustedOracleFunctionEnum;
 
 /***
  * oracle datasource dialect.
@@ -50,9 +51,15 @@ public class OracleDatabaseDialect extends AbstractDatabaseDialect {
 	public String getLimitPageSqlWithOffset(String sql, int startOffset, int pageSize) {
 		return sql + "  OFFSET " + startOffset + " ROWS FETCH NEXT " + pageSize + " ROWS ONLY ";
 	}
-	
+
 	@Override
 	public String getLimitPageSql(String sql, int pageNo, int pageSize) {
 		return sql + "  OFFSET " + getPagePrevNum(pageNo, pageSize) + " ROWS FETCH NEXT " + pageSize + " ROWS ONLY ";
+	}
+
+
+	@Override
+	public String getFunction(String functionName) {
+		return TrustedOracleFunctionEnum.getFunctionByName(functionName);
 	}
 }

--- a/nacos-datasource-plugin-ext/nacos-oracle-datasource-plugin-ext/src/main/java/com/alibaba/nacos/plugin/datasource/enums/TrustedOracleFunctionEnum.java
+++ b/nacos-datasource-plugin-ext/nacos-oracle-datasource-plugin-ext/src/main/java/com/alibaba/nacos/plugin/datasource/enums/TrustedOracleFunctionEnum.java
@@ -1,0 +1,67 @@
+/*
+ * Copyright 1999-2018 Alibaba Group Holding Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.alibaba.nacos.plugin.datasource.enums;
+
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * The TrustedSqlFunctionEnum enum class is used to enumerate and manage a list of trusted built-in SQL functions.
+ * By using this enum, you can verify whether a given SQL function is part of the trusted functions list
+ * to avoid potential SQL injection risks.
+ *
+ * @author blake.qiu
+ */
+public enum TrustedOracleFunctionEnum {
+
+    /**
+     * NOW().
+     */
+    NOW("NOW()", "SYSTIMESTAMP");
+
+    private static final Map<String, TrustedOracleFunctionEnum> LOOKUP_MAP = new HashMap<>();
+
+    static {
+        for (TrustedOracleFunctionEnum entry : TrustedOracleFunctionEnum.values()) {
+            LOOKUP_MAP.put(entry.functionName, entry);
+        }
+    }
+
+    private final String functionName;
+
+    private final String function;
+
+    TrustedOracleFunctionEnum(String functionName, String function) {
+        this.functionName = functionName;
+        this.function = function;
+    }
+
+    /**
+     * Get the function name.
+     *
+     * @param functionName function name
+     * @return function
+     */
+    public static String getFunctionByName(String functionName) {
+        TrustedOracleFunctionEnum entry = LOOKUP_MAP.get(functionName);
+        if (entry != null) {
+            return entry.function;
+        }
+        throw new IllegalArgumentException(String.format("Invalid function name: %s", functionName));
+    }
+}

--- a/nacos-datasource-plugin-ext/nacos-oracle-datasource-plugin-ext/src/main/java/com/alibaba/nacos/plugin/datasource/impl/oracle/AbstractOracleMapper.java
+++ b/nacos-datasource-plugin-ext/nacos-oracle-datasource-plugin-ext/src/main/java/com/alibaba/nacos/plugin/datasource/impl/oracle/AbstractOracleMapper.java
@@ -21,6 +21,7 @@ import com.alibaba.nacos.common.utils.NamespaceUtil;
 import com.alibaba.nacos.common.utils.StringUtils;
 import com.alibaba.nacos.plugin.datasource.constants.DatabaseTypeConstant;
 import com.alibaba.nacos.plugin.datasource.dialect.DatabaseDialect;
+import com.alibaba.nacos.plugin.datasource.enums.TrustedOracleFunctionEnum;
 import com.alibaba.nacos.plugin.datasource.manager.DatabaseDialectManager;
 import com.alibaba.nacos.plugin.datasource.mapper.AbstractMapper;
 
@@ -174,5 +175,10 @@ public abstract class AbstractOracleMapper extends AbstractMapper {
 
 	protected DatabaseDialect getDatabaseDialect() {
 		return databaseDialect;
+	}
+
+	@Override
+	public String getFunction(String functionName) {
+		return databaseDialect.getFunction(functionName);
 	}
 }

--- a/nacos-datasource-plugin-ext/nacos-postgresql-datasource-plugin-ext/src/main/java/com/alibaba/nacos/plugin/datasource/dialect/PostgresqlDatabaseDialect.java
+++ b/nacos-datasource-plugin-ext/nacos-postgresql-datasource-plugin-ext/src/main/java/com/alibaba/nacos/plugin/datasource/dialect/PostgresqlDatabaseDialect.java
@@ -17,6 +17,7 @@
 package com.alibaba.nacos.plugin.datasource.dialect;
 
 import com.alibaba.nacos.plugin.datasource.constants.DatabaseTypeConstant;
+import com.alibaba.nacos.plugin.datasource.enums.TrustedPostgresqFunctionEnum;
 
 /**
  * PostgreSQL database dialect.
@@ -47,6 +48,11 @@ public class PostgresqlDatabaseDialect extends AbstractDatabaseDialect {
     @Override
     public String getLimitPageSqlWithOffset(String sql, int startOffset, int pageSize){
         return sql + "  OFFSET " + startOffset + " LIMIT " + pageSize;
+    }
+
+    @Override
+    public String getFunction(String functionName) {
+        return TrustedPostgresqFunctionEnum.getFunctionByName(functionName);
     }
 
 }

--- a/nacos-datasource-plugin-ext/nacos-postgresql-datasource-plugin-ext/src/main/java/com/alibaba/nacos/plugin/datasource/enums/TrustedPostgresqFunctionEnum.java
+++ b/nacos-datasource-plugin-ext/nacos-postgresql-datasource-plugin-ext/src/main/java/com/alibaba/nacos/plugin/datasource/enums/TrustedPostgresqFunctionEnum.java
@@ -1,0 +1,66 @@
+/*
+ * Copyright 1999-2018 Alibaba Group Holding Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.alibaba.nacos.plugin.datasource.enums;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * The TrustedSqlFunctionEnum enum class is used to enumerate and manage a list of trusted built-in SQL functions.
+ * By using this enum, you can verify whether a given SQL function is part of the trusted functions list
+ * to avoid potential SQL injection risks.
+ *
+ * @author blake.qiu
+ */
+public enum TrustedPostgresqFunctionEnum {
+
+    /**
+     * NOW().
+     */
+    NOW("NOW()", "NOW()");
+
+    private static final Map<String, TrustedPostgresqFunctionEnum> LOOKUP_MAP = new HashMap<>();
+
+    static {
+        for (TrustedPostgresqFunctionEnum entry : TrustedPostgresqFunctionEnum.values()) {
+            LOOKUP_MAP.put(entry.functionName, entry);
+        }
+    }
+
+    private final String functionName;
+
+    private final String function;
+
+    TrustedPostgresqFunctionEnum(String functionName, String function) {
+        this.functionName = functionName;
+        this.function = function;
+    }
+
+    /**
+     * Get the function name.
+     *
+     * @param functionName function name
+     * @return function
+     */
+    public static String getFunctionByName(String functionName) {
+        TrustedPostgresqFunctionEnum entry = LOOKUP_MAP.get(functionName);
+        if (entry != null) {
+            return entry.function;
+        }
+        throw new IllegalArgumentException(String.format("Invalid function name: %s", functionName));
+    }
+}

